pkg/solana/config/chain_reader.go

```diff
@@ -0,0 +1,37 @@
+package config
+
+import "github.com/smartcontractkit/chainlink-common/pkg/codec"
+
+type ChainReader struct {
+	Namespaces map[string]ChainReaderMethods `json:"namespaces" toml:"namespaces"`
+}
+
+type ChainReaderMethods struct {
+	Methods map[string]ChainDataReader `json:"methods" toml:"methods"`
+}
+
+type ChainDataReader struct {
+	AnchorIDL  string                 `json:"anchorIDL" toml:"anchorIDL"`
+	Procedures []ChainReaderProcedure `json:"procedures" toml:"procedures"`
+}
+
+type ProcedureType int
+
+const (
+	ProcedureTypeInternal ProcedureType = iota
+	ProcedureTypeAnchor
+)
+
+type ChainReaderProcedure chainDataProcedureFields
+
+type chainDataProcedureFields struct {
+	// IDLAccount refers to the account defined in the IDL.
+	IDLAccount string `json:"idlAccount"`
+	// Type describes the procedure type to use such as internal for static values,
+	// anchor-read for using an anchor generated IDL to read values from an account,
+	// or custom structure for reading from a native account.
```

Comment:
I don't think you support the custom structure yet.  It might be worth saying that the currently supported value is anchor-read, but that others will be added in future for X

Reply:
I can do that. More explanation would help in usability.